Vizzuality/marxan-cloud
# [MARXAN-62] tighten container security (api, geoprocessing)

### Overview

Clean up and tighten security of api and geoprocessing images

* fix `addgroup`/`adduser`
* run build commands as non-root user
* tidy up `COPY` commands in `Dockerfile`
* run container as non-root user (set at runtime in `docker-compose.yml`, needs to be set accordingly if running outside of docker compose)

### Designs

N/A

### Testing instructions

* the three images affected should build
* the API and geoprocessing services should start
* running the API in CI E2E test (`make test-e2e-api`) should run tests
* ~geodata import should work~ (we'll address this in a separate PR)
* files in `/opt/marxan-api` should be owned by `marxan-api:marxan-api` (API container)
* files in `/opt/marxan-geoprocessing` should be owned by `marxan-geoprocessing:marxan-geoprocessing` (geoprocessing container)
* running `ps xau` in a running container should show the node/make/etc processes running as a non-root user

### Feature relevant tickets

https://vizzuality.atlassian.net/browse/MARXAN-62

---

## Checklist before submitting

      deploying to staging/production, please add brief testing instructions
      to the deploy checklist (`docs/deployment-checklist.md`)
